packages/scaffold-config/src/ct-detect-third-party.ts

```diff
@@ -123,10 +131,12 @@ export async function detectThirdPartyCTFrameworks (
     if (packageJsonPaths.length === 0) {
       debug('no third-party dependencies detected')
 
-      return []
+      return { frameworks: [], erroredFrameworks }
     }
 
     debug('found third-party dependencies %o', packageJsonPaths)
+    let name = ''
+    let modulePath = ''
```

Comment:
This feels dangerous to me, but maybe my `let` context scoping wrt promises isn't up to snuff. This would seem to share state variables across multiple promises and seems like a potential leakage issue (state initialized in one promise overwrites and is used as a result for a second based on invocation order & how the promises yield to one another).

Reply:
This is a good thing to point out. Kinda want to look into it a little bit, I think you're right about the risk in a general sense, maybe this is too simplistic.